src/urllib3/fields.py

```diff
@@ -12,7 +13,7 @@
     cast,
 )
 
-_TYPE_FIELD_VALUE = Union[str, bytes]
+_TYPE_FIELD_VALUE = Union[str, bytes, BinaryIO]
```

Comment:
The docstring for all of this shows using a file here so this was missing but it had the knock on effect of rewriting the header params which were overloading the definition of a "FIELD_VALUE"

Reply:
Hmm, I'm not sure if files were supported. Iirc it was `open (x).read()`? Need to check on PC

Reply:
Hm, so I guess requests always papered over this: https://github.com/psf/requests/blob/d09659997cd1e3eca49a07c59ece5557071c0ab9/requests/models.py#L160

I don't know how we consolidate these APIs though in such a way that folks don't need to read the entire file into memory.